include/zephyr/sys/util.h

```diff
@@ -546,6 +546,55 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 		(expr);                                                                            \
 	})
 
+/** @brief Number of bits that make up a type */
+#define NUM_BITS(t) (sizeof(t) * 8)
+
+/**
+ * @brief Compute the log2 of an unsigned integer
+ *
+ * @param x value to compute logarithm of (positive only)
+ *
+ * @returns log2(x)
+ */
+#define LOG2(x)                                                                                    \
```

Comment:
This does not handle x=0, which is undefined in `__builtin_clz()`. IMO it should either be supported or mentioned in the macro docs.